services/members.js

```diff
@@ -13,3 +13,15 @@ exports.getMembers = async () => {
     throw new ErrorObject(error.message, error.statusCode || 500)
   }
 }
+
+exports.deleteMember = async (id) => {
+  try {
+    const member = await Member.destroy({ where: { id } })
+    if (!member) {
```

Comment:
como esta marcando german en otros pr... para mejorar el código.. ya no seria necesario devolver este error xq si se rompe el destroy tiraria el error por el catch.. tendria que esperar la promesa y si la cumple hacer return.. sino .. tira el error por el catch

Reply:
+1
Salvo que tengas una validacion especifica sobre la respuesta de la db, el catch se va a encargar de absorver la respuesta no positiva de la db

Reply:
Yo también pensé lo mismo que ustedes, pero probando me di cuenta de que el catch no absorbe el caso en el cual no se encuentra el id. Por lo tanto, si es necesario tener ese condicional.

Adjunto pruebas
En este caso el id 3 **no existe**

Con el condicional:
![image](https://user-images.githubusercontent.com/74195715/191278103-ff683b54-a5a2-4579-827d-18f416f96678.png)

Sin el condicional:
![image](https://user-images.githubusercontent.com/74195715/191278142-58a81882-77c7-41a0-9fd6-18dc1e8a0dfa.png)

Reply:
aca me parece que falta algo.. falta validar que el usuario existe con la id que viene por params, y si lo encuentra... eliminarlo

Reply:
Entiendo lo que querés decir, sería buscar el id con `.findByPk` y a la respuesta hacerle el `.destroy`.

El tema es que, si findByPk no encuentra el elemento, devuelve null (y no un error), por lo que del mismo modo se debería hacer un condicional para comprobar esto.
En el caso de tratar de hacer destroy al elemento null (en caso de que findByPk no encuentre el elemento) se obtiene un error mucho menos descriptivo "I`nternalServerError: [Error deleting member] - [members - DELETE]: Cannot read properties of null`" con el código 500 (internal server error)

Por esto considero que mi solución es la más simple y descriptiva. Ya que en caso de no encontrar el miembro devuelve el error "`NotFoundError: [Error deleting member] - [members - DELETE]: No member found`" con un código 404